src/Components/Patient/PatientConsentRecords.tsx

```diff
@@ -281,29 +269,36 @@ export default function PatientConsentRecords(props: {
               </>
             )}
           </div>
+          <div className="mt-2 text-sm text-red-500">{fileUpload.error}</div>
         </div>
         <div className="flex-1">
-          <div className="flex flex-col gap-4">
-            {loading ? (
+          {consentRecords?.filter(
+            (r) =>
+              files?.results.filter((f) => f.associating_id === r.id).length,
+          ).length === 0 ? (
+            <div className="flex h-32 items-center justify-center text-gray-500">
+              No consent records found
+            </div>
+          ) : (
+            (!unarchivedFiles || !archivedFiles) &&
+            !consentRecords && (
               <div className="skeleton-animate-alpha h-32 rounded-lg" />
-            ) : tabConsents?.length === 0 || !filesFound ? (
-              <div className="flex h-32 items-center justify-center text-gray-500">
-                No records found
-              </div>
-            ) : null}
-            {!loading &&
-              tabConsents?.map((record, index) => (
-                <PatientConsentRecordBlockGroup
-                  key={index}
-                  consentRecord={record}
-                  previewFile={fileManager.viewFile}
-                  archiveFile={fileManager.archiveFile}
-                  onDelete={(record) => setShowDeleteConsent(record.id)}
-                  refreshTrigger={consultation}
-                  showArchive={showArchived}
-                  onFilesFound={() => setFilesFound(true)}
-                />
-              ))}
+            )
+          )}
+          <div className="flex flex-col gap-4">
+            {consentRecords?.map((record, index) => (
+              <PatientConsentRecordBlockGroup
+                key={index}
+                consentRecord={record}
+                previewFile={fileManager.viewFile}
+                archiveFile={fileManager.archiveFile}
+                editFile={fileManager.editFile}
+                showArchive={showArchived}
+                files={files?.results.filter(
```

Comment:
Explore the possibility to Associate Files to consentRecord State, rather than using a separate state for consent and files